[browser-kit] dev-lang/spidermonkey Autogening

org-tekeli-borisp opened the pull request.

org-tekeli-borisp committed:
[browser-kit] dev-lang/spidermonkey Update patch URIs to macaronios.org.

Replaced Gentoo developer-hosted patch URIs with macaronios.org equivalents for improved hosting reliability and availability. This change ensures proper access to required patchsets during the build process.

(cherry picked from commit 3b9537d2d855b105eb616b085b13987ac64d6379)
(cherry picked from commit 96012141c0e3bb5c4e65d5deff7eec3d49bb0a4f)

org-tekeli-borisp committed:
[browser-kit] dev-lang/spidermonkey Autogen dev-lang/spidermonkey and delete obsolete static dev-lang/spidermonkey ebuilds.

(cherry picked from commit d4dfc55be26d0bd8019e239a5be81d422e744f38)
(cherry picked from commit ee0282b5d51341812e128d247879e29bd6b0c7c3)

org-tekeli-borisp committed:
[browser-kit] dev-lang/spidermonkey Remove unused 'files' field from SpiderMonkey autogen.yaml

The 'files' field was redundant and is no longer used in the build process. Cleaning up the autogen file helps maintain clarity and avoids potential confusion or errors in future updates.

(cherry picked from commit cdab87da545031a036235e79eb1702c8cde456b5)
(cherry picked from commit 996bd2ec7fffb2a3d130b14d0f94afd4bc6a3ccb)

org-tekeli-borisp committed:
Update LLVM and Clang dependencies in templates

This update aligns the LLVM and Clang dependencies in SpiderMonkey and Firefox templates to versions 16. This ensures the use of the updated toolchains, thereby removing the dependencies on older versions. The maximum supported LLVM slot has also been adjusted accordingly. This modification ensures compatibility with the build of ISPC and addresses any discrepancies that may arise with newer version releases.

(cherry picked from commit 8be011b43b12bc5b7609cb50ab7752c8a6787b43)
(cherry picked from commit 00c5c41154416be23dafce8d73bdc1ddcb611aa4)

org-tekeli-borisp committed:
[browser-kit] dev-lang/spidermonkey Update spidermonkey template to fix LTO dependency handling

Added a missing dependency constraint for LTO builds when using clang. This ensures proper compatibility and prevents build issues related to linker requirements.

(cherry picked from commit 06a7f3072fd3c881f9f2b4a4796dd5167beda9eb)
(cherry picked from commit df2b713189dfae681544044e44370e7eb50c7f0b)

org-tekeli-borisp committed:
[browser-kit] dev-lang/spidermonkey Remove Manifest file for dev-lang/spidermonkey

The Manifest file for dev-lang/spidermonkey was deleted as it is no longer needed. This helps reduce redundancy and maintain a cleaner repository.

geaaru commented on browser-kit/curated/dev-lang/spidermonkey/autogen.yaml:
```diff
@@ -0,0 +1,15 @@
+        homepage: https://developer.mozilla.org/en-US/docs/Mozilla/Projects/SpiderMonkey
+        license: MPL-2.0
+        version: 78.15.0
+        revision: 2
```
Maybe you can add:
```
llvm_max_slot: 16
```
and later use {{ llvm_max_slot }} in the template?
This will permit to easily update the template.

org-tekeli-borisp replied:
good idea

geaaru commented on lang-kit/curated/dev-lang/ispc/templates/ispc.tmpl:
```diff
@@ -17,7 +17,7 @@ LICENSE="BSD BSD-2 UoI-NCSA"
 IUSE="examples"
 
-RDEPEND="<sys-devel/clang-13:="
+RDEPEND="<sys-devel/clang-17:="
```
This seems wrong from the beginning.
The `clang` package is slotted. It's correct to have a specific version but i dunno if it's correct block installation of other packages.
This could be replaced maybe with:
```
RDEPEND="sys-devel/clang:16"
```
?

geaaru replied:
This will permit probably to remove `llvm` eclass and `LLVM_MAX_SLOT` variable.

org-tekeli-borisp replied:
removed ISPC from PR  It requires at least LLVM 17!

geaaru replied:
I mean that if you use SLOT.
```
sys-devel/clang:16
```
your system will require CLANG 16 but will support system also with additional slots. The difference is that it permits to install LLVM 17/19 but it needs CLANG 16.

Are there conflicts file of the ISPC with LLVM 17?

org-tekeli-borisp replied:
![image](https://github.com/user-attachments/assets/043389bc-15e8-4cc8-82c0-60d8c482ad7f)

```
Would you like to merge these packages? [Yes/No] 
>>> Verifying ebuild manifests
>>> Emerging (1 of 1) dev-lang/ispc-1.26.0::lang-kit
>>> Failed to emerge dev-lang/ispc-1.26.0, Log file:
>>>  '/run/portage/dev-lang/ispc-1.26.0/temp/build.log'
>>> Jobs: 0 of 1 complete, 1 failed                 Load avg: 0.83, 1.23, 1.55
 * Package:    dev-lang/ispc-1.26.0
 * Repository: lang-kit
 * USE:        amd64 elibc_glibc kernel_linux userland_GNU
 * FEATURES:   network-sandbox preserve-libs sandbox userpriv usersandbox
 * Using python3.9 to build
>>> Unpacking source...
>>> Unpacking ispc-1.26.0.tar.gz to /run/portage/dev-lang/ispc-1.26.0/work
>>> Source unpacked in /run/portage/dev-lang/ispc-1.26.0/work
>>> Preparing source in /run/portage/dev-lang/ispc-1.26.0/work/ispc-1.26.0 ...
 * Removing auto-x86 build on amd64
 * Working in BUILD_DIR: "/run/portage/dev-lang/ispc-1.26.0/work/ispc-1.26.0_build"
 * Hardcoded definition(s) removed in CMakeLists.txt:
 *      set(CMAKE_BUILD_TYPE "Release")
 * Hardcoded definition(s) removed in examples/cpu/CMakeLists.txt:
 *      set(CMAKE_BUILD_TYPE "Release")
 * Hardcoded definition(s) removed in examples/xpu/CMakeLists.txt:
 *      set(CMAKE_BUILD_TYPE "Release")
>>> Source prepared.
>>> Configuring source in /run/portage/dev-lang/ispc-1.26.0/work/ispc-1.26.0 ...
 * Working in BUILD_DIR: "/run/portage/dev-lang/ispc-1.26.0/work/ispc-1.26.0_build"
cmake -C /run/portage/dev-lang/ispc-1.26.0/work/ispc-1.26.0_build/gentoo_common_config.cmake -G Ninja -DCMAKE_INSTALL_PREFIX=/usr -DARM_ENABLED=no -DCMAKE_SKIP_RPATH=ON -DISPC_NO_DUMPS=ON -DISPC_INCLUDE_EXAMPLES=no -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_TOOLCHAIN_FILE=/run/portage/dev-lang/ispc-1.26.0/work/ispc-1.26.0_build/gentoo_toolchain.cmake  /run/portage/dev-lang/ispc-1.26.0/work/ispc-1.26.0
CMake Warning:
  Ignoring empty string ("") provided on the command line.


loading initial cache file /run/portage/dev-lang/ispc-1.26.0/work/ispc-1.26.0_build/gentoo_common_config.cmake
-- The C compiler identification is GNU 12.3.0
-- The CXX compiler identification is GNU 12.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/x86_64-pc-linux-gnu-gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/x86_64-pc-linux-gnu-g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- LLVM_CONFIG_EXECUTABLE: /usr/lib/llvm/16/bin/llvm-config
-- LLVM_TOOLS_BINARY_DIR: /usr/lib/llvm/16/bin
-- LLVM_DIR is /usr/lib/llvm/16/bin/../lib/cmake/llvm
-- CLANG_EXECUTABLE: /usr/lib/llvm/16/bin/clang
-- CLANGPP_EXECUTABLE: /usr/lib/llvm/16/bin/clang++
-- LLVM_AS_EXECUTABLE: /usr/lib/llvm/16/bin/llvm-as
-- FILE_CHECK_EXECUTABLE: /usr/lib/llvm/16/bin/FileCheck
-- Detected LLVM version: 16.0.6
-- GIT_BINARY: /usr/bin/git
fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
-- M4 macro processor: /usr/bin/m4
-- Found Python3: /run/portage/dev-lang/ispc-1.26.0/temp/python3.9/bin/python3 (found version "3.9.21") found components: Interpreter 
-- Found BISON: /usr/bin/bison (found suitable version "3.8.2", minimum required is "3.0") 
-- Found FLEX: /usr/bin/flex (found suitable version "2.6.4", minimum required is "2.6") 
CMake Error at CMakeLists.txt:371 (message):
  ISPC requires LLVM 17.0.0 or later


-- Configuring incomplete, errors occurred!
See also "/run/portage/dev-lang/ispc-1.26.0/work/ispc-1.26.0_build/CMakeFiles/CMakeOutput.log".
 * ERROR: dev-lang/ispc-1.26.0::lang-kit failed (configure phase):
 *   cmake failed
 * 
 * Call stack:
 *     ebuild.sh, line   93:  Called src_configure
 *   environment, line 3285:  Called cmake_src_configure
 *   environment, line 1329:  Called die
 * The specific snippet of code:
 *       "${CMAKE_BINARY}" "${cmakeargs[@]}" "${CMAKE_USE_DIR}" || die "cmake failed";
 * 
 * If you need support, post the output of `emerge --info '=dev-lang/ispc-1.26.0::lang-kit'`,
 * the complete build log and the output of `emerge -pqv '=dev-lang/ispc-1.26.0::lang-kit'`.
 * The complete build log is located at '/run/portage/dev-lang/ispc-1.26.0/temp/build.log'.
 * The ebuild environment file is located at '/run/portage/dev-lang/ispc-1.26.0/temp/environment'.
 * Working directory: '/run/portage/dev-lang/ispc-1.26.0/work/ispc-1.26.0_build'
 * S: '/run/portage/dev-lang/ispc-1.26.0/work/ispc-1.26.0'
 ```

org-tekeli-borisp committed:
[browser-kit] dev-lang/spidermonkey Set llvm_max_slot as a configurable parameter

Previously, llvm_max_slot was hardcoded in the template. This change moves it to the autogen.yaml file, allowing for better configurability and future flexibility in managing versions.

geaaru commented on browser-kit/curated/dev-lang/spidermonkey/templates/spidermonkey.tmpl:
```diff
@@ -72,24 +60,10 @@ BDEPEND="${PYTHON_DEPS}
 			clang? (
-				sys-devel/clang:10
-				lto? ( =sys-devel/lld-10* )
+				sys-devel/clang:16
```
Can you use `{{ llvm_max_slot }}`  or `{{ llvm_slot }}` here too and in the other reference to CLANG/LLVM?
Same in firefox.tmpl?

org-tekeli-borisp replied:
done

org-tekeli-borisp committed:
[browser-kit] dev-lang/spidermonkey Update llvm dependencies to use llvm_max_slot variable

Replaced hardcoded LLVM slot numbers with the llvm_max_slot variable in the spidermonkey template. This ensures flexibility and consistency when handling different LLVM versions. The change simplifies future maintenance of the dependency definitions.

geaaru commented on browser-kit/curated/dev-lang/spidermonkey/templates/spidermonkey.tmpl:
```diff
@@ -72,24 +60,10 @@ BDEPEND="${PYTHON_DEPS}
 			clang? (
-				sys-devel/clang:10
-				lto? ( =sys-devel/lld-10* )
+				sys-devel/clang:{ llvm_max_slot }}
```
typo ;) 
Miss a `{`

org-tekeli-borisp replied:
done!

org-tekeli-borisp committed:
[browser-kit] dev-lang/spidermonkey Update llvm dependencies to use llvm_max_slot

Replaced hardcoded LLVM slot values with the {{ llvm_max_slot }} variable in the spidermonkey template. This improves flexibility and consistency in managing LLVM versions and reduces the likelihood of maintenance errors in the future.
